### PR TITLE
Infer Stylus parser in Vue SFC

### PR DESCRIPTION
**Pull request by lsdsjy**
## Description

Allow prettier to recognize style[lang="stylus"] block in Vue SFC.
Though prettier doesn't support stylus now, users could write a custom stylus plugin for prettier. However they cannot make it work with style[lang="stylus"] block in Vue SFC without this change.
And I believe it won't break anything.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
This is not how it works, is there a stylus plugin? We can support it.

**Comment by fisker** [comment]
We need to have that plugin loaded, and search in `prettier.getSupportInfo().languages` to find the parser name.

**Comment by lsdsjy** [comment]
Sorry if I didn't make it clear.
Yes, I am working on a prettier plugin for the stylus language. When I run prettier like

```
prettier --plugin . tests/example.styl
```
It can format the .styl file well. But if I want to prettify the `<style lang="stylus">` block in Vue SFC like

```
prettier --plugin . tests/example-with-stylus.vue
```
It won't work. So I took a look at the SCSS/Less support in Vue SFCs and thus this PR.

Of course it'd be great if we can make prettier support stylus. But after I looked at the project, I think it'll be hard to integrate stylus support into prettier. Stylus is a very flexible language, for example, colons, semicolons, commas and braces all can be omitted. If we shall allow all these styles, we should provide many new options in prettier, while 
 as I know prettier is very conservative in adding options. So in a more pragmatic way I choose to write a prettier plugin for my own usage, keep it simple without many customizations, and make it work with Vue SFCs well.

**Comment by fisker** [comment]
As I said, the solution should be https://github.com/prettier/prettier/pull/12707#issuecomment-1110799720

**Comment by lsdsjy** [comment]
@fisker Sorry I didn't really get you. Patched.

**Comment by fisker** [comment]
Looks great, have you tested?

**Comment by fisker** [comment]
We should add a simple test for it. If you are not familiar with the code, I can do it for you.

**Comment by fisker** [comment]
Good, can you fix the linting issue and add a changelog?

File: tests/config/prettier-plugins/prettier-plugin-dummy-stylus/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      parse: (text) => ({ text }),
+      astFormat: astName,
+      locStart() {},
+      locEnd() {},
+    },
+  },
+  printers: {
+    [astName]: {
+      print(path) {
+        return path.getValue().text + "\n// by dummy parser";
```
Comment (fisker) [inline tests/config/prettier-plugins/prettier-plugin-dummy-stylus/index.js:24]: I think this has to be a fixed string, second run will add one more comment , I guess the CI will fail.

File: tests/config/prettier-plugins/prettier-plugin-dummy-stylus/index.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    },
+  },
+  printers: {
+    [astName]: {
+      print(path) {
+        return path.getValue().text.replace(/;/g, "");
+      },
+    },
+  },
+};
```
Comment (fisker) [inline tests/config/prettier-plugins/prettier-plugin-dummy-stylus/index.js:28]: Can you simiply this plugin with this util?

**Review by fisker — APPROVED** [review]
Good job

File: src/language-html/utils/index.js [hunk 3]
```diff
@@ -386,6 +386,10 @@ function inferStyleParser(node) {
   if (lang === "less") {
     return "less";
   }
+
+  if (lang === "stylus") {
+    return inferParserByLanguage("stylus", options);
+  }
```
Comment (sosukesuzuki) [inline src/language-html/utils/index.js:392]: Can you add a comment that explains why `"stylus"` is here? e.g.:

```suggestion
  // Prettier does not officially support stylus.
  // But, we need to handle `"stylus"` here for printing a style block in Vue SFC as stylus code by external plugin.
  // https://github.com/prettier/prettier/pull/12707
  if (lang === "stylus") {
    return inferParserByLanguage("stylus", options);
  }
```

Comment (lsdsjy) [reply src/language-html/utils/index.js:395]: Makes Sense. Done.

**Review by sosukesuzuki — APPROVED** [review]

**Comment by lsdsjy** [comment]
I think I can add a test case with a dummy plugin for stylus and run it on a vue file.

**Pull request merged**